### ISSUE TITLE
Add a representation of load shedding to the demand flexibility model

### ISSUE THREAD

**Issue by lanesmith** [Breakthrough-Energy/REISE.jl#125]
Currently, all demand flexibility is implemented as "shiftable demand", where all curtailed demand must be recovered. However, flexible demand can also be classified as "sheddable demand", where curtailed demand does not have to be recovered. Sheddable demand is common in incentive-based demand response programs. Such a representation would be useful, especially considering the demand response with human-in-the-loop collaboration.

Implementing such a model would likely be easy, especially with much of the code infrastructure in place from the current shiftable demand model. Sheddable demand would be very similar, with the exceptions that (1) it would not need load balancing constraints and (2) it would not be able to be price responsive like the shiftable demand (due to the cost minimization, the sheddable demand would want to curtail its maximum amount every time step). For sheddable demand to adequately participate in the model, there will need to be a different representation of costs and benefits, as discussed in #124.

NOTE: Although load shedding does already exist in the model, this load shedding would be different (though not too different mathematically). While the current representation of load shedding is used mostly for feasibility purposes (e.g., extreme line congestion preventing a simulation from being feasible), the proposed load shedding discussed in this issue would be used to model the impact that demand response programs can have on the system.

**Comment by merrielle** [comment]
@lanesmith Do we still need to work on this?

**Comment by lanesmith** [comment]
> @lanesmith Do we still need to work on this?

Yeah, it's still not implemented. In the current way we model demand flexibility, which is solely through load shifting, load shedding would not work as intended because it would simply shed all demand that it could. Under the current framework, we will need to have costs associated with the load shedding so that shed decisions are more reasonable. We have a basic cost parameter associated with the demand flexibility model, but this is something that might change as our demand-response-with-human-in-the-loop collaborators continue to explore costs associated with demand flexibility. Unless we need this for something else, I would probably recommend holding off on this feature until the cost structure is better determined, just so we don't need to refactor too much in the future. However, if needed, it probably wouldn't be too difficult to get a basic model of load shedding implemented with the current cost framework.
